src/Sylius/Bundle/CoreBundle/spec/OAuth/UserProviderSpec.php

```diff
@@ -223,7 +224,7 @@ function it_should_throw_exception_when_no_email_was_provided(
         $response->getEmail()->willReturn(null);
         $userRepository->findOneByEmail(Argument::any())->shouldNotBeCalled();
 
-        $this->shouldThrow(UsernameNotFoundException::class)->during('loadUserByOAuthUserResponse', [$response]);
+        $this->shouldThrow(UserNotFoundException::class)->during('loadUserByOAuthUserResponse', [$response]);
```

Comment:
So we're missing a test here to be able to opt-in. This is a BC break.

Reply:
The longer I think about it, I'm more convinced it is not. Adding class in the middle is the backward compatible change - https://github.com/SyliusLabs/PolyfillSymfonySecurity/blob/master/src/Core/Exception/UserNotFoundException.php. Therefore, we are free to do it. As long as you are on Sf5.4 your old code will work with an update here. Once you will bump to Sf6.0, the class will change but the previous won't exist anyway, so you had to adjust class anyway.

The valid point, you brought is that people may be using a new exception in Sf5.4 - but that's not exactly the case for us in my opinion. First of all, usage of the new exception will not work (as we are triggering the old one anyway). If you want to prepare your codebase for the next Symfony version you still have at least two options:
```php
try {

} catch (Symfony\Component\Security\Core\Exception\UsernameNotFoundException|Symfony\Component\Security\Core\Exception\UserNotFoundException) {}
```
or
```php
try {

} catch (SyliusLabs\Polyfill\Symfony\Security\Core\Exception\UserNotFoundException) {}
```

which will behave quite similar.

My doubts about the new argument are that we would need to maintain this parameter for the next major version. In addition, it will be problematic to remove it in backward compatible way